lib/reverse_words.rb

```diff
@@ -1,3 +1,7 @@
+# Share and explain the time and space complexities for your solution.
+# The time complexity is O(n) because it's two loops that are linear so it's O(n) + O(n)
+# The space complexity is O(1) because it's outputting the same amount as input
+
 def reverse_word(string, array)
```

Comment:
For clarity i.e. easier readability and maintainability, I'd recommend passing the starting index and ending index of the word as two additional separate parameters rather than in an array of size 2 called string.